packages/trace-viewer/src/index.tsx

```diff
@@ -26,12 +26,11 @@ import { WorkbenchLoader } from './ui/workbenchLoader';
   if (window.location.protocol !== 'file:') {
     if (window.location.href.includes('isUnderTest=true'))
       await new Promise(f => setTimeout(f, 1000));
-    navigator.serviceWorker.register('sw.bundle.js');
-    if (!navigator.serviceWorker.controller) {
-      await new Promise<void>(f => {
-        navigator.serviceWorker.oncontrollerchange = () => f();
-      });
-    }
+    await navigator.serviceWorker.register('sw.bundle.js');
+    // navigator.serviceWorker.controller is null for a force refresh, hence
+    // we do a soft refresh to be controlled by the service worker again.
+    if (!navigator.serviceWorker.controller)
```

Comment:
There is no relation between the service worker installation / activation and its registration, so the old code was good.